src/shared.js

```diff
@@ -121,8 +121,8 @@ export function foldLine (str, maxLen = 76) {
 
     // ensure that the line never ends with a partial escaping
     // make longer lines if needed
-    while (curLine.substring(-1) === '\\' && pos + curLine.length < len) {
-      curLine += str.charAt(pos + curLine.length);
+    while (curLine.endsWith('\\') && pos + curLine.length < len) {
+      curLine += str.charAt(pos + curLine.length + 1); // Append the next character
```

Comment:
@erikyo I don't think this is correct, it should be the original because `pos` would be considered an inclusive part of the string, it includes the first character of `curLine`.

Reply:
I ask because that part written that way never works, or at least I have never been able to make that while condition come true and run the "line currying"

Reply:
It is a weird function, I'll give it a deeper look.

Reply:
Just throwing it out there, I think the intention was to prevent escaped characters from breaking, but I believe this is already implicit since we refer to the value of the string and not the way it is "written".... It's impossible for a line to break between `\"` just to be clear.

This is a test that shows the "issue"
```js
it('should ensure that the line never ends with a partial escaping', () => {
  const line = 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa\'aaaaa\'aaaa';
  const folded = foldLine(line);

  expect(line).to.equal(folded.join(''));
  expect(folded).to.deep.equal([
    "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'",
    "aaaaa'aaaa"
  ]);
});
```

Reply:
@erikyo think you are right, that once the string is created it doesn't actually have an "escape" character taking up a position in the string.

Reply:
So the whole concept of including the escape character is flawed, even if the string ends in a literal `\` the next character isn't considered an escape character.

Reply:
followup here 3f65da7ea33499339e2191434258bf5940e42449